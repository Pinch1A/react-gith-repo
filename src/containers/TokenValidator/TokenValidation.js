import * as actions from '../../store/actions/index'
import React, { useEffect, useState } from 'react'
import { Button } from '../../App'
import { connect } from 'react-redux'

import styled from 'styled-components'

const TokenPage = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 4rem;
`
const Header = styled.div`
  margin-bottom: 4rem;
  text-align: center;
  font-size: 1.5rem;
`

export const Title = styled.h3`
  font-size: 3rem;
  font-weight: bold;
  margin-bottom: 2rem;
`
const TextField = styled.input`
  padding: 1rem;
  border: 1px solid grey;
  border-radius: 1rem;
  width: 30rem;
  margin-bottom: 2rem;
  margin: 20px 0;
`
const FormWrapper = styled.form`
  display: flex;
  flex-direction: column;
`

const Error = styled.h4`
  color: salmon;
  font-size: 2rem;
`

const TokenValidator = ({
  tokenValidator,
  error,
  history,
  token: existingToken,
}) => {
  const [token, setToken] = useState(existingToken)
  const [errorMessage, setErrorMessage] = useState(
    error ? error.message : undefined
  )

  const handleForm = (e) => {
    e.preventDefault()
    tokenValidator({ token, history: history })
  }
  const inputChangeHandler = (e) => {
    setToken(e.target.value)
  }
  const handleFormWithEnv = () => {
    tokenValidator({
      token: process.env.REACT_APP_GITHUB_PERSONAL_ACCESS_TOKEN,
      history: history,
    })
  }

  useEffect(() => {
    if (error === 'INVALID') setErrorMessage('Token is InValid')
  }, [error, errorMessage])

  return (
    <TokenPage>
      <Header>
        <Title>Token Validator</Title>
        <p>
          Before Proceeding further please validate your token. To generate
          token and find more info refer{' '}
          <a
            rel="noopener noreferrer"
            href="https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-tokens"
            target="_blank"
          >
            <u>docs</u>
          </a>
          .
        </p>
      </Header>

      <Button primary onClick={handleFormWithEnv}>
        USE ENVIRONMENT VAR
      </Button>
      <FormWrapper autoComplete="off" onSubmit={handleForm}>
        <TextField
          type="text"
          name="token"
          onChange={inputChangeHandler}
          placeholder="Enter your token here....."
        />
        {errorMessage && <Error>{errorMessage}</Error>}
      </FormWrapper>
    </TokenPage>
  )
}

const mapStateToProps = (state) => {
  return {
    token: state.tokenValidator.token,
    loading: state.tokenValidator.loading,
    error: state.tokenValidator.error,
  }
}
const mapDispatchToProps = (dispatch) => {
  return {
    tokenValidator: ({ token, history }) =>
      actions.tokenValidator({ token, history })(dispatch),
  }
}
export default connect(mapStateToProps, mapDispatchToProps)(TokenValidator)
