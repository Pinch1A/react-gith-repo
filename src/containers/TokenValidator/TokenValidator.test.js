import TokenValidator, { Title } from './TokenValidation'
import { mount, shallow } from 'enzyme'
import { Provider } from 'react-redux'
import React from 'react'
import configureStore from 'redux-mock-store'

const mockStore = configureStore([])

const mockTokenValidator = (data) => {
  return {
    token: 'token-test',
    loading: false,
    error: null,
    ...data,
  }
}

describe('<ReposPage>', () => {
  let store

  beforeEach(() => {
    store = mockStore({
      tokenValidator: mockTokenValidator(),
    })
  })
  it('renders without crashing', () => {
    const props = {
      ...mockTokenValidator(),
      history: jest.fn(),
    }

    const wrapper = shallow(
      <Provider store={store}>
        <TokenValidator {...props} />
      </Provider>
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('shows title', () => {
    const props = {
      ...mockTokenValidator(),
      history: jest.fn(),
    }

    const wrapper = mount(
      <Provider store={store}>
        <TokenValidator {...props} />
      </Provider>
    )
    expect(wrapper.find(Title).text()).toEqual('Token Validator')
  })
})
