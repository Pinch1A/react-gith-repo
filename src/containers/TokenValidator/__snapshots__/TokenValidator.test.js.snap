// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReposPage> renders without crashing 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(TokenValidator)
    error={null}
    history={[MockFunction]}
    loading={false}
    token="token-test"
  />
</ContextProvider>
`;
